addons/account/models/account_move.py

```diff
@@ -6127,7 +6127,11 @@ def is_internal_partner(partner):
             'invoice_source_email': from_mail_addresses[0],
             'partner_id': partners and partners[0].id or False,
         }
-        move_ctx = self.with_context(default_move_type=custom_values['move_type'], default_journal_id=custom_values['journal_id'])
+        move_ctx = self.with_context(
+            default_move_type=custom_values['move_type'],
+            default_journal_id=custom_values['journal_id'],
+            default_company_id=custom_values.get('company_id', company)
```

Comment:
Because of:
`company = self.env['res.company'].browse(custom_values['company_id']) if custom_values.get('company_id') else self.env.company`

```suggestion
            default_company_id=company.id,
```